p2p/peer.go

```diff
@@ -130,6 +130,12 @@ func (puc *PeerUpdatesCh) Done() <-chan struct{} {
 	return puc.doneCh
 }
 
+// TestSend performs a blocking send of a PeerUpdate on the PeerUpdatesCh
+// updatesCh. Note, this should be used for testing purposes only.
+func (puc *PeerUpdatesCh) TestSend(pu PeerUpdate) {
```

Comment:
@erikgrinaker Let me know how you feel about this.

It will probably be pretty common for isolated reactor tests to need the ability to directly send `PeerUpdate`s. I've added this function for testing purposes only (hence the name).

We can either:

1. Keep and use this method.
2. Turn `PeerUpdatesCh` into an interface (where we use a dummy/bidirectional channel for testing purposes).
3. Remove the read-only semantics from `PeerUpdatesCh`.

I prefer to stick with (1) or (2).

Reply:
Why don't we just use the same patterns here as with `Channel`, i.e. take the channel in `NewPeerUpdatesCh`? Then the test can send on it all it wants.

Reply:
Yeah I didn't think of that -- we can modify `NewPeerUpdates` to take a channel instead of internally constructing one.